.github/workflows/ci/pkg/environment/environment.go

```diff
@@ -111,22 +118,94 @@ func New(c Config) (*PullRequestEnvironment, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	// Check if the pull request has changes to the `docs/` directory as that will affect who
+	// the required reviewers are.
+	docChanges, codeChanges, err := hasChanges(c.Context, pr, c.Client)
+	if err != nil {
+		// Log the error, don't fail the whole run because it couldn't detect if the pull request has docs.
+		log.Errorf("error while detecting pull request for docs changes: %v, skipping assigning docs reviewers", err)
+	}
+
 	return &PullRequestEnvironment{
 		Client:           c.Client,
 		reviewers:        c.Reviewers,
 		defaultReviewers: c.Reviewers[ci.AnyAuthor],
 		Metadata:         pr,
+		HasDocsChanges:   docChanges,
+		HasCodeChanges:   codeChanges,
 	}, nil
 }
 
+// hasChanges determines if the pull request has docs and/or code changes.
+func hasChanges(ctx context.Context, pr *Metadata, clt *github.Client) (hasDocsChanges bool, hasCodeChanges bool, err error) {
+	files, err := getPullRequestFiles(ctx, pr, clt)
+	if err != nil {
+		return false, true, trace.Wrap(err)
+	}
+	for _, file := range files {
+		if file.Filename == nil {
+			return false, true, trace.BadParameter("pull request file name is nil")
+		}
```

Comment:
When would this happen? Would that PR just get no one assigned to review it?

Reply:
I'm not sure when it would happen (this is type from the `go-github`), just being safe and checking if it's nil. 

Correct, no one would get assigned, the workflow would fail, and we would need to investigate the issue. Maybe I should just use `continue` instead of returning an error.

Reply:
I deleted the previous comment cause I forgot I logged the error instead of returning -- If the file name was nil and `hasDocsChanges` and `hasCodeChanges` were both false, we would still assign the required reviewers. 

Also not sure when the filename would be nil, but just checking to be safe so nothing panics.